proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -4531,40 +4531,67 @@ lemma refill_budget_check_round_robin_cur_sc_active[wp]:
   "refill_budget_check_round_robin consumed \<lbrace>cur_sc_active\<rbrace>"
   by (rule cur_sc_active_lift; wpsimp)
 
-method handle_overrun_loop_body_simple
-  = ((clarsimp simp: handle_overrun_loop_body_def)?
-     , wpsimp wp: whileLoop_valid_inv set_refills_wp get_refills_wp
-            simp: refill_pop_head_def vs_all_heap_simps obj_at_def refill_single_def refill_size_def
-                  update_sched_context_set_refills_rewrite update_refill_hd_rewrite
-                  schedule_used_defs)
+method charge_entire_head_refill_simple
+  = (clarsimp simp: charge_entire_head_refill_def)?,
+    wpsimp wp: whileLoop_valid_inv set_refills_wp get_refills_wp
+         simp: refill_pop_head_def vs_all_heap_simps obj_at_def refill_single_def refill_size_def
+               update_sched_context_set_refills_rewrite update_refill_hd_rewrite
+               schedule_used_defs sc_at_ppred_def
 
-method handle_overrun_loop_simple
-  = ((clarsimp simp: handle_overrun_loop_def)?
-     , wpsimp wp: whileLoop_valid_inv set_refills_wp get_refills_wp
-            simp: handle_overrun_loop_body_def merge_overlapping_refills_def round_robin_def refill_pop_head_def
-                  sc_valid_refills_def vs_all_heap_simps obj_at_def refill_budget_check_defs
-                  update_sched_context_set_refills_rewrite schedule_used_defs)
+method handle_overrun_simple
+  = (clarsimp simp: handle_overrun_def)?,
```

Comment:
this looks nearly identical to the previous method, am I missing why we need two methods?

Reply:
They are slightly different. It's possible that they could be combined into one method that could be used instead of the two, but it could be rather annoying to try this.